api/v1/common/sagemaker_api.go

```diff
@@ -408,7 +408,7 @@ type ObjectiveStatus string
 // Hosting related types
 
 type ProductionVariant struct {
-	AcceleratorType ProductionVariantAcceleratorType `json:"acceleratorType,omitempty"`
+	AcceleratorType *string `json:"acceleratorType,omitempty"`
```

Comment:
Why not make them `*ProductionVariantAcceleratorType` ? These should be identical

Reply:
`string` seems more straight forward

Reply:
Yes but we lose specificity when we make this change. We have lots of other types in this file that alias string, even `ProductionVariantInstanceType` is still an alias. This change isn't relevant to Go SDK v1 migration, so I would like to keep it the same.

Reply:
* yes we have other alias which are `string` but none of them are passed as pointers `*alias`
  * In this file we have 18 string type variables and all are passed as string but never as string pointer 
* All `*string` variables are passed as is and they do not have aliases 
  * In this file we have 79 `*string` variables all of which are passed as `*string` and they don't have alias
* This change is relevant to Go SDK v1 migration because `ProductionVariantAcceleratorType` went from being a `string` to `*string`. This datatype is used in the controllers.

So this change of removing `ProductionVariantAcceleratorType` and using `*string` is consistent with what we currently have in this file

Reply:
Absolutely fair points. Thank you for explaining this to me